app/models/product.rb

```diff
@@ -0,0 +1,9 @@
+class Product < ApplicationRecord
+  has_many :category_products, dependent: :destroy
+  has_many :categories, through: :category_products
+
+  validates :name, :description, :price, :rating, :status, :stock, presence: true
+  validates :rating, :status, :stock, numericality: { greater_than: 0 }
+
+  enum status: { inactive: 0, active: 1 }
```

Comment:
The data supplied in the dump, these values are `0` and `1` for a large volume of records. So, in order not to change all these values, the best thing to do is to do it with an enum and it gives the possibility that in the future, if another state is required, this will support it.